### PR TITLE
Dynamic payment schema for polygon

### PR DESCRIPTION
**Pull request by scx1332**

### CODE REVIEW

File: core/payment-driver/erc20/src/erc20/ethereum.rs [hunk 1]
```diff
@@ -44,16 +49,38 @@ pub fn get_polygon_starting_price() -> f64 {
...
+pub fn get_polygon_max_gas_price_dynamic() -> f64 {
+    return std::env::var("POLYGON_MAX_GAS_PRICE_DYNAMIC")
+        .unwrap_or("1000.0".to_string())
+        .parse::<f64>()
+        .unwrap_or(1000.0);
+}
+
+pub fn get_polygon_gas_price_method() -> PolygonGasPriceMethod {
+    match std::env::var("POLYGON_GAS_PRICE_METHOD")
+        .unwrap_or("fast".to_string())
```
Comment (tworec) [inline core/payment-driver/erc20/src/erc20/ethereum.rs:77]: `fast` will be treated as dynamic below. Is it on purpose?

File: core/payment-driver/base/src/dao/transaction.rs [hunk 2]
```diff
@@ -236,6 +236,33 @@ impl<'c> TransactionDao<'c> {
...
+            TransactionStatus::Confirmed => Some(current_time),
+            _ => None,
+        };
+        do_with_transaction(self.pool, move |conn| {
+            diesel::update(dsl::transaction.find(tx_id))
+                .set((
+                    dsl::status.eq(status as i32),
+                    dsl::time_last_action.eq(current_time),
+                    dsl::time_confirmed.eq(confirmed_time),
+                    dsl::last_error_msg.eq(err),
```
Comment (tworec) [inline core/payment-driver/base/src/dao/transaction.rs:257]: 👍

**Comment by tworec** [comment]
@scx1332 it would be great if you can describe the strategy introduced within this PR in few sentences (currently description is void, and it took me several minutes to get to the point)

**Review by nieznanysprawiciel — DISMISSED** [review]
If we need to execute different code for each network, than probably we should make an abstraction (dyn Trait) for this.

But this is not blocking comment.

**Review by nieznanysprawiciel — COMMENTED** [review]
Please check my comments, before I will approve

File: core/payment-driver/erc20/src/erc20/wallet.rs [hunk 3]
```diff
@@ -194,32 +210,43 @@ pub async fn make_transfer(
 fn bump_gas_price(gas_in_gwei: f64, limit_in_gwei: Option<f64>) -> f64 {
     let mut result = -1.0;
 
-    let polygon_prices = get_polygon_priority();
+    match get_polygon_gas_price_method() {
+        PolygonGasPriceMethod::PolygonGasPriceDynamic => {
+            //ignore maximum gas price, because we have to bump at least 10% so the transaction will be accepted
+            result = gas_in_gwei * 1.11;
+            result
+        }
```
[the remark this answers is not in the document]
Comment (tworec) [reply core/payment-driver/erc20/src/erc20/wallet.rs]: imo no, see elswhere, there are comparisons with max_gas

**Review by prekucki — APPROVED** [review]

**Review by nieznanysprawiciel — DISMISSED** [review]

**Comment by maaktweluit** [comment]
we are getting a lof of settings around gas.. would be nice to make it simpler instead of adding layers on top of layers of fixes

**Review by nieznanysprawiciel — APPROVED** [review]

**Pull request merged**